internal/store/serverservice/serverservice.go

```diff
@@ -222,69 +223,92 @@ func (r *Store) AssetUpdate(ctx context.Context, asset *model.Asset) error {
 				"hr": hr,
 			}).Warn("server service server query returned nil object")
 
-		return errors.Wrap(ErrServerServiceQuery, "got nil Server object")
+		return errors.Wrap(model.ErrInventoryQuery, "got nil Server object")
 	}
 
-	// create/update inventory.
-	if errInventory := r.createUpdateInventory(ctx, asset, server); errInventory != nil {
+	// publish server inventory
+	if errPublishInv := r.publishInventory(ctx, asset, server); errPublishInv != nil {
 		r.logger.WithFields(
 			logrus.Fields{
 				"id":  id,
 				"hr":  hr,
-				"err": errInventory.Error(),
-			}).Warn("inventory asset insert/update error")
+				"err": errPublishInv.Error(),
```

Comment:
just use `logrus.WithError` here.

Reply:
I gave that a shot, although since `WithError` only accepts the error as a parameter, we'd have to drop the `id` (serverID), `hr` (HSS response) fields which provide a more context on the response received.

Reply:
You can chain the methods: ```r.logger.WithError(errPublishInv).WithFields(logrus.Fields{…}).Warn(…)```

Reply:
TIL, thanks